Tapi 2 5 0 fixes

nigel-r-davis opened the pull request:
Protobuf file added #619 
Corrections to deal with config false issue #620

nigel-r-davis committed:
Adding Protobuf files

As agreed in issue #619

nigel-r-davis committed:
Config false issue #620

The following have been adjusted to be require-instance false;

In tapi-connectivity.yang
- connection-end-point-ref
- switch-control-ref
- switch-ref
- route-ref

In tapi-common.yang
- profile-ref

In tapi-topology.yang
- topology-ref
- node-ref

rcasellas reviewed (approved):
good for me

roshan-joyce-fujitsu commented on PROTOBUF/tapi-streaming-performance.proto:
```diff
@@ -0,0 +1,137 @@
+     string maintenance_intermediate_point = 13 [json_name = "maintenance-intermediate-point"];
+     string maintenance_end_point = 14 [json_name = "maintenance-end-point"];
+     uint64 measurement_start_time = 15 [json_name = "measurement-start-time"];
+     repeated QualifiedMeasuredValue qualified_measured_value_set = 16 [json_name = "qualified-measured-value-set"];
```
I think we need to add the following here:
```
repeated NameAndValue additionalInfo = 17 [json_name = "additional-info"];
```

And define `NameAndValue` as:
```
message NameAndValue {
    string valueName = 1 [json_name = "value-name"];
    string value = 2;
}
```

This will be required to be consistent with the yang model in tapi-gnmi-streaming.yang.

nigel-r-davis replied:
I have made the two modifications proposed. I have added to the pull request.

I will also need to make the changes in the 2.7 deliverable.

roshan-joyce-fujitsu commented on PROTOBUF/tapi-streaming-performance.proto:
```diff
@@ -0,0 +1,137 @@
+     string maintenance_intermediate_point = 13 [json_name = "maintenance-intermediate-point"];
+     string maintenance_end_point = 14 [json_name = "maintenance-end-point"];
+     uint64 measurement_start_time = 15 [json_name = "measurement-start-time"];
+     repeated QualifiedMeasuredValue qualified_measured_value_set = 16 [json_name = "qualified-measured-value-set"];
```
I think we need to add the following here also:
```
repeated NameAndValue additionalInfo = 17 [json_name = "additional-info"];
```
This will be required to be consistent with the yang model in tapi-gnmi-streaming.yang.

nigel-r-davis replied:
I have made the modification proposed. I have added to the pull request. I will also need to make the changes in the 2.7 deliverable.

roshan-joyce-fujitsu reviewed (commented):
Hi @nigel-r-davis 

Only 2 comments to make it consistent with the yang model in tapi-gnmi-streaming.yang.

Otherwise, looks good.

nigel-r-davis committed:
Modifications to the protobuf to align with the YANG definitions

repeated NameAndValue additionalInfo = 17 [json_name = "additional-info"];

The above has been added in two places and the definition below has also been added.

message NameAndValue {
    string valueName = 1 [json_name = "value-name"];
    string value = 2;
}

roshan-joyce-fujitsu reviewed (approved):
Looks good
